### ISSUE TITLE
less usable stations :( 

### ISSUE THREAD

**Issue by Hannah-Bee** [hg1337/homeassistant-dwd#3]
My impression is that the number of usable stations is very small. Any MOSMIX listed station of the DWD in the radius of my location, which according to the list provide a lot of data (even the relevant ones) can not be used with the integration. 
And using a station that is more than 40km away relativizes the added value. Too bad. Would love to use the integration.

**Comment by hg1337** [comment]
Hi @Hannah-Bee,

first of all, sorry for the very late reply.

We recently added support for 330 additional stations that had IDs that unfortunately were not recognised by this integration before. Please check, if this helps. If there are still stations that are not supported but should be, please provide some examples (IDs and maybe also the names of the stations) so I can have a deeper look into this.

Thanks, @hg1337

**Comment by dasGOLE** [comment]
I have the same issue, the closest station for me is N7316 MERZIG and J709 MERZIG [ID, City] and both are not working.
And another issue I got is, that I can't figure out, how I get the Precipitation from any station. I can only select "Precipitation unit" as an Attribute. O,o
![image](https://user-images.githubusercontent.com/110224654/211407970-f3cd39b1-0e92-474e-aa19-e4d242e0de30.png)

**Comment by hg1337** [comment]
Thank you @dasGOLE, I will have a look into the stations.

But I can already answer regarding the precipitation: It's only supported for forecasts, not as a measurement of the current weather following the documentation of the WeatherEntity: https://developers.home-assistant.io/docs/core/entity/weather/ (Notice that the native_precipitation_unit is still on entity level so that it's not repeated for each forecast, but the native_precipitation is only on forecast level.)

But for the forecasts, it should be there if the station supports it (i.e. if RR1c is present).

![image](https://user-images.githubusercontent.com/76538974/211412962-061c1e54-41bf-49b5-a1ea-ea88f634b215.png)

**Comment by hg1337** [comment]
Regarding the two stations N7316 and J709, unfortunately, they are both not listed at https://opendata.dwd.de/weather/weather_reports/poi/ which means the they do not provide measurement data. As some measurements (condition and native_temperature) are mandatory attributes of the WeatherEntity, only stations that provide measurement data are currently supported. (And I'm not so much a fan of selling the (usually not so fresh) forecast for the current time as measurements, even if it's only as a fallback, because it's quite misleading what is being provided which will lead to other complaints.) That's basically also what the error message says when you manually enter the station:

![image](https://user-images.githubusercontent.com/76538974/211415987-a329629a-488b-425a-9b55-757e54f45fb8.png)

**Comment by dasGOLE** [comment]
Thank you for the fast response!
What makes me curious, the weather station of Merzig worked fine for this (https://github.com/FL550/dwd_weather) other dwd weather integration. But as long as I can figure out, how I can use the perception forecast, both integration should work for me. I look in to that later.

**Comment by hg1337** [comment]
They don't use measurement data at all, they always use the forecast for the current hour, see [README.md](https://github.com/FL550/dwd_weather/blob/master/README.md).

I'm going to look into how I can maybe use this as a fallback without making it too confusing.

**Comment by Hannah-Bee** [comment]
@hg1337 Thank you for your adjustments. The weather station I need is now recognized and provides data. I will start experimenting with it now and give feedback if necessary. But for now it looks very promising now

**Comment by hg1337** [comment]
@Hannah-Bee I'm happy that is looks good for you so far. :) To be precise, @mpg07 did the necessary adjustments.

@dasGOLE I have added support for stations without measurement data, check out the latest release.

![image](https://user-images.githubusercontent.com/76538974/212522273-6194e969-d1f0-4b35-b569-d1456b6a9d9c.png)

However, I'm not sure how reliable the stations without measurement data are generally. When picking some randomly, I quickly found one that had completely weird forecasts, like 30°C etc. (should rather have been like 10°C, but I looked into the data, it was there.). :o But maybe this was just bad luck.
